presto-main/src/main/java/com/facebook/presto/operator/DriverYieldSignal.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+public class DriverYieldSignal
+{
+    @GuardedBy("this")
+    private long runningSequence;
+
+    @GuardedBy("this")
+    private ScheduledFuture<?> scheduledFuture;
+
+    @GuardedBy("this")
+    private boolean yieldScheduled;
```

Comment:
I don't think this variable is needed anymore.  It looks like `yiedScheduled == scheduledFurture == null` is true through the code.